UnityProject/Assets/Scripts/Items/Renameable.cs

```diff
@@ -0,0 +1,80 @@
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+public class Renameable : Interactable<HandActivate>, IRightClickable
+{
+	public NetTabType NetTabType = NetTabType.Rename;
+
+	protected override bool WillInteract(HandActivate interaction, NetworkSide side)
+	{
+		if (!base.WillInteract(interaction, side)) return false;
+
+		var cnt = GetComponent<CustomNetTransform>();
+		var ps = interaction.Performer.GetComponent<PlayerScript>();
+		var pna = interaction.Performer.GetComponent<PlayerNetworkActions>();
+
+		if (pna.GetActiveHandItem() == gameObject)
+		{
+			return true;
+		}
+
+		if (!ps.IsInReach(cnt.RegisterTile, side == NetworkSide.Server))
+		{
+
+			return false;
+		}
+
+		return true;
+	}
+
+	protected override void ServerPerformInteraction(HandActivate interaction)
+	{
+		OpenRenameDialog(interaction.Performer);
+	}
+
+	public RightClickableResult GenerateRightClickOptions()
+	{
+		var result = RightClickableResult.Create();
+
+		if (WillInteract(HandActivate.ByLocalPlayer(), NetworkSide.Client))
+		{
+			result.AddElement("Rename", RightClickInteract);
+		}
+
+		return result;
+	}
+
+	private void RightClickInteract()
+	{
+		Interact(HandActivate.ByLocalPlayer(), nameof(Renameable));
+	}
+
+	private void OpenRenameDialog(GameObject player)
+	{
+		TabUpdateMessage.Send(player, gameObject, NetTabType, TabAction.Open);
+	}
+
+	public void SetCustomName(string msg)
+	{
+		var attributes = gameObject.GetComponent<ItemAttributes>();
+		var customName = attributes.itemName.Split('-')[0].Trim();
```

Comment:
won't this mess with items that contain `-` in their name?

Reply:
Assumably, yes! This was supposed to get rid of the existing custom name, and most likely this component won't be on anything that does have - in it's name, but for future-proofing and doing things safely, the custom name and original name are stored here and synced.